### DISCUSSION TITLE
Error Running PySR with Julia in Windows 10 (Anaconda Environment)

### DISCUSSION THREAD

**Discussion by sbalasbas3** [MilesCranmer/PySR#585] [Q&A] [answered]
**Environment Details:**

- Operating System: Windows-10-10.0.19045-SP0
- Python Version: 3.11.7
- Julia Version: (Unfortunately, I wasn't able to retrieve this detail due to the current issue, but it's supposed to be compatible with the latest pysr and juliacall versions.)
- juliacall Version: 0.9.19
- PySR Version: 0.18.0

**My Problem:**

Hello, I'm encountering an error when trying to use PySR. The initialization of PySR fails. Despite installing and setting up PySR and Julia, as well as ensuring that the juliacall package is up to date, I run into a CalledProcessError during the execution of Julia package management commands initiated by juliacall. I have no idea how to resolve this issue. This error occurs despite following the setup procedures for PySR and Julia within an Anaconda environment on Windows 10.

`CalledProcessError: Command '['C:\\Users\\[USERNAME]\\AppData\\Local\\anaconda3\\envs\\sym_reg\\julia_env\\pyjuliapkg\\install\\bin\\julia.exe', '--project=C:\\Users\\[USERNAME]\\AppData\\Local\\anaconda3\\envs\\sym_reg\\julia_env', '--startup-file=no', '-e', 'import Pkg; Pkg.Registry.update(); Pkg.add([Pkg.PackageSpec(name="PythonCall", uuid="6099a3de-0909-46bc-b1f4-468b9a2dfc0d"), Pkg.PackageSpec(name="SymbolicRegression", uuid="8254be44-1295-4e6a-a16d-46603ac705cb", url=raw"https://github.com/MilesCranmer/SymbolicRegression.jl", rev=raw"v0.24.1"), Pkg.PackageSpec(name="Serialization", uuid="9e88b42a-f829-5b0c-bbe9-9e923198166b")]); Pkg.resolve(); Pkg.precompile()']' returned non-zero exit status 1.`

**Steps taken so far:**

- Ensured juliacall is up to date by running `pip install --upgrade juliacall`, which confirmed version 0.9.19 is installed.
- Attempted to manually specify the Julia executable's path using the JULIA_PYTHONCALL_EXE environment variable before importing PySR, without success. I wrote the code like this: `julia_path = "C:\\Users\\[USERNAME]\\AppData\\Local\\anaconda3\\envs\\sym_reg\\julia_env\\pyjuliapkg\\install\\bin\\julia.exe"` and `os.environ["JULIA_PYTHONCALL_EXE"] = julia_path`
- Activated Julia Environment: I activated the specific Julia project environment located at `C:\Users\[USERNAME]\AppData\Local\anaconda3\envs\sym_reg\julia_env` using the `Pkg.activate` command, ensuring that all package operations were localized to this environment. After activating the environment, I executed `Pkg.instantiate()` to install and update dependencies as defined in the environment's Project.toml and Manifest.toml files. This command updated the registry and added the PythonCall v0.9.19 and SymbolicRegression v0.24.1 packages among others. I also used `Pkg.add()` to explicitly add `PythonCall`, `SymbolicRegression` and `Serialization`.

**Comment by gm89uk** [comment]
Hi  @sbalasbas3, I had this issue following updating to Pysr 18.0, and spent the evening trying to fix it. 
https://github.com/MilesCranmer/PySR/issues/267#issuecomment-1741268870
This fix worked for me. 
1) C:\Users\<your username>\anaconda3\julia_env\pyjuliapkg\install\bin\julia.exe, I opened the REPL and followed the instructions. 

Good luck.

**Comment by MilesCranmer** [comment]
The main difference in 0.18.0 is it installs SymbolicRegression.jl from the git repo rather than from the Julia registry. Perhaps some machines don't have the right LibGit libraries. Let me revert that change and push a patch.

The change was made as sometimes the registries are out-of-date. But I guess there are worse effects like this one.

**Comment by MilesCranmer** [comment]
Hoping https://github.com/MilesCranmer/PySR/pull/587 will fix this.

**Comment by MilesCranmer** [comment, marked as answer]
Fixed in 0.18.1, just releasing now. Give Conda ~5 hours to update, or install the pip version. Can you verify it is fixed?

**Reply by sbalasbas3** [reply]
Thank you very much for this! I have updated PySR to version 0.18.1 using pip and the issue has been resolved.
